### PR TITLE
Testing support for Apollo Datasources

### LINKED ISSUE #201: Add Testing Support for Apollo Datasources
## User Story

As a `<type of user>` _(who)_ I want to `<perform an action>` _(what)_ so that I can `<achieve a goal>` _(why)_.

#### Problem Details _(Optional)_

* Screenshots, GIFs, or a video
* Any additional information (version, device, etc.)

#### Acceptance Criteria _(Optional)_

* [ ] It should [SPEC_DETAIL]

### PR DESCRIPTION
**Pull request by vinnyjth**
Fixes #201

## Creator

- [x] Build relevant tests if any apply
- [x] Ensure there are no new warnings (in app AND in storybook)
- [ ] Upload an animated GIF showcasing the solution (if it applies)
- [x] Set a relevant reviewer
- [ ] Add a line to the changelog documenting the high-level change this PR introduces

## Reviewer

- [ ] Review CI build results
- [ ] Review function and form on iOS
- [ ] Review function and form on Android
- [ ] Review new stories if any apply
- [ ] Review test coverage: Are all new or changed states tested?
- [ ] Review new test logic
- [ ] Review the changelog entry

### CODE REVIEW

File: api/__mocks__/apollo-server-env.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import fetch from 'jest-fetch-mock';
+
+import * as apolloDatasourceMocks from './apollo-datasource-mocks';
+
+const apollo_server_env = require.requireActual('apollo-server-env');
```
Comment (IsaacHardy) [inline api/__mocks__/apollo-server-env.js]: Doesn't look like our eslint setup observes these files, but it looks like this is a camelCase lint issue here.

File: api/__mocks__/apollo-server-env.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+const nodeFetch = require.requireActual('node-fetch');
```
Comment (IsaacHardy) [inline api/__mocks__/apollo-server-env.js]: Not sure if intended, but nodeFetch is never used.

**Comment by vinnyjth** [comment]
Thanks @IsaacHardy  :D

File: api/src/index.js [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
+import { resolvers, schema } from './data';
 
 import getContext from './getContext';
+import getDataSources from './getDataSources';
 
 export default new ApolloServer({
   typeDefs: schema,
   resolvers,
-  dataSources: () => ({ liveStream: new dataSources.LiveStream() }),
+  dataSources: getDataSources,
```
Comment (IsaacHardy) [inline api/src/index.js]: I like this.

File: api/src/data/live/__tests__/resolver.tests.js [hunk 4]
```diff
@@ -1,34 +1,34 @@
...
-    //     liveStream {
-    //       isLive
-    //     }
-    //   }
-    // `;
-    // const rootValue = {};
+    const query = `
+      query {
+        liveStream {
+          isLive
```
[the remark this answers is not in the document]
Comment (vinnyjth) [reply api/src/data/live/__tests__/resolver.tests.js]: Good call :D

[the remark this answers is not in the document]
Comment (IsaacHardy) [reply api/src/data/live/__tests__/resolver.tests.js]: I don't see any changes to the query.

[the remark this answers is not in the document]
Comment (vinnyjth) [reply api/src/data/live/__tests__/resolver.tests.js]: Look at the line below :)

[the remark this answers is not in the document]
Comment (IsaacHardy) [reply api/src/data/live/__tests__/resolver.tests.js]: I think Github had a hiccup.

**Pull request merged**